packages/nouns-subgraph/src/nouns-dao.ts

```diff
@@ -57,60 +62,95 @@ import {
+  proposal.maxQuorumVotesBPS = dynamicQuorum.maxQuorumVotesBPS;
+  proposal.quorumCoefficient = dynamicQuorum.quorumCoefficient;
+
+  proposal.clientId = 0;
+  proposal.save();
+
+  captureProposalVersion(
+    event.transaction.hash.toHexString(),
+    event.logIndex.toString(),
+    proposal,
+    false,
+  );
+}
+
 export function handleProposalCreatedWithRequirements(
   event: ProposalCreatedWithRequirements1,
 ): void {
-  handleProposalCreated(ParsedProposalV3.fromV1Event(event));
+  handleParsedProposal(ParsedProposalV3.fromV1Event(event));
 }
 
 export function handleProposalCreatedWithRequirementsV3(
   event: ProposalCreatedWithRequirements,
 ): void {
-  handleProposalCreated(ParsedProposalV3.fromV3Event(event));
+  handleParsedProposal(ParsedProposalV3.fromV3Event(event));
 }
 
-export function handleProposalCreatedOnTimelockV1(event: ProposalCreatedOnTimelockV1): void {
-  let proposal = getOrCreateProposal(event.params.id.toString());
-  proposal.onTimelockV1 = true;
-  proposal.save();
+export function handleProposalCreatedWithRequirementsV4(
+  event: ProposalCreatedWithRequirementsV4,
+): void {
+  handleParsedProposal(ParsedProposalV3.fromV4Event(event));
 }
 
-export function handleProposalCreated(parsedProposal: ParsedProposalV3): void {
+export function handleParsedProposal(parsedProposal: ParsedProposalV3): void {
   let proposal = getOrCreateProposal(parsedProposal.id);
-  let proposerResult = getOrCreateDelegateWithNullOption(parsedProposal.proposer);
-
-  // Check if the proposer was a delegate already accounted for, if not we should log an error
-  // since it shouldn't be possible for a delegate to propose anything without first being 'created'
-  if (proposerResult.created && parsedProposal.signers.length == 0) {
-    log.error('Delegate {} not found on ProposalCreated. tx_hash: {}', [
-      parsedProposal.proposer,
-      parsedProposal.txHash,
-    ]);
-  }
 
-  // Create it anyway, which supports V3 cases of proposers not having any Nouns
-  proposal.proposer = proposerResult.entity!.id;
-  proposal.targets = parsedProposal.targets;
-  proposal.values = parsedProposal.values;
-  proposal.signatures = parsedProposal.signatures;
-  proposal.calldatas = parsedProposal.calldatas;
-  proposal.createdTimestamp = parsedProposal.createdTimestamp;
-  proposal.createdBlock = parsedProposal.createdBlock;
-  proposal.lastUpdatedTimestamp = parsedProposal.createdTimestamp;
-  proposal.lastUpdatedBlock = parsedProposal.createdBlock;
-  proposal.createdTransactionHash = parsedProposal.createdTransactionHash;
-  proposal.startBlock = parsedProposal.startBlock;
-  proposal.endBlock = parsedProposal.endBlock;
-  proposal.updatePeriodEndBlock = parsedProposal.updatePeriodEndBlock;
-  proposal.proposalThreshold = parsedProposal.proposalThreshold;
-  proposal.quorumVotes = parsedProposal.quorumVotes;
   proposal.forVotes = BIGINT_ZERO;
   proposal.againstVotes = BIGINT_ZERO;
   proposal.abstainVotes = BIGINT_ZERO;
```

Comment:
is setting the votes to zero needed here? isn't it already done in `handleProposalCreated`?

Reply:
because events are handled out of order on goldsky, we need to choose between setting them to zero in both function, vs making these values nullable as well. I chose to just set them again because we know they must be zero at this point. Making them nullable felt more annoying. wdyt?